src/Avalonia.Base/Rendering/Composition/CompositingRenderer.cs

```diff
@@ -172,7 +172,8 @@ private void UpdateCore()
                 v.SynchronizeCompositionChildVisuals();
         _dirty.Clear();
         _recalculateChildren.Clear();
-        CompositionTarget.Size = _root.ClientSize;
+        
+        CompositionTarget.PixelSize = PixelSize.FromSizeRounded(_root.ClientSize, _root.RenderScaling);
```

Comment:
In some backends, like Browser, we can get this pixel size without doing roundtrip of scaling conversion. See https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry/devicePixelContentBoxSize

Possibly, toplevel itself should report its pixel size?

Reply:
Yes, but I've decided against touching every single backend including ones I can't really test.

Reply:
I meant more moving PixelSize.FromSizeRounded to the TopLevel level, instead of updating each backend at once. It can be done gradually, once main API is there.

Reply:
`devicePixelContentBoxSize` itself returns pixel size that can be used to set render size of the html canvas. 
Since our ClientSize is scaled, right now we divide devicePixelContentBoxSize by scaling factor. 
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Browser/Avalonia.Browser/AvaloniaView.cs#L473

Reply:
Can be done in separated PR